### DISCUSSION TITLE
Errors after moving container manager: Drive btrfs failed to remove root file system

### DISCUSSION THREAD

**Discussion by stevenhaskell** [007revad/Synology_app_mover#121] [Q&A] [answered]
I was able to successfully move Container Manager to a new volume using the script. Most containers were able to run without an issue. A few of them have an error message along the lines below. Any idea how to resolve this? 

> container 7e851076ddc391b22a2f380c79f1baca9319337c3d4b01ae0ec5c27cf5b99653:
> "driver "btrfs" failed to remove root filesystem: Failed to destroy btrfs snapshot
> /volume2/@docker/btrfs/subvolumes for
> 27da9bf15376d3f61ae63702d5e41497aa1d49d5d9e97240115c8cea503a30c4: invalid argument

**Comment by 007revad** [comment]
Do you have backups of the json or yml files for the affected containers?

I've grown to hate container manager or docker as it causes 80% of the issues people report having with syno_ap_mover.

Does the following command return anything?
```
ls /volume2/@docker/btrfs/subvolumes | grep  grep -E '^27da9b'
```

**Reply by stevenhaskell** [reply]
@007revad Yes, I do. I created manual backups and backups through your script. I ran the command and it returned the following:

```
grep: ^27da9b: No such file or directory
ls: cannot access '/volume2/@docker/btrfs/subvolumes': Permission denied
```

**Reply by 007revad** [reply]
Is `/volume2/@docker` empty?
```
sudo ls /volume2/@docker/btrfs/subvolumes
```

**Reply by stevenhaskell** [reply]
No, it's showing a huge wall of subvolumes. I can see the one in the original error message. Some do end with "-init". Example below:
```
27da9bf15376d3f61ae63702d5e41497aa1d49d5d9e97240115c8cea503a30c4       
27da9bf15376d3f61ae63702d5e41497aa1d49d5d9e97240115c8cea503a30c4-init
```

**Comment by 007revad** [comment]
You could try removing dangling docker images:
```
sudo docker image prune --force
```
or even removing dangling and unused docker images:
```
sudo docker image prune --all --force
```
Then delete the affected containers and their images, and recreate them from your backups.

**Reply by stevenhaskell** [reply]
I'll give this a shot and report back. Thanks for your help on this!

**Reply by stevenhaskell** [reply, marked as answer]
@007revad sorry for the delay. The command you recommended didn't work. I was able to remove it after force deleting the sub volume mentioned. Once I did, I could then delete it in container manager.
`sudo rm -rf /volume2/@docker/btrfs/subvolumes/replaceWithSubVolumeName`

Thank you for your help anyways!
